Sources/StreamChat/Models/Attachments/ChatMessageAttachment.swift

```diff
@@ -47,6 +54,18 @@ public extension ChatMessageAttachment {
 extension ChatMessageAttachment: Equatable where Payload: Equatable {}
 extension ChatMessageAttachment: Hashable where Payload: Hashable {}
 
+/// A type represeting the downloading state for attachments.
+public struct AttachmentDownloadingState: Hashable {
+    /// The local file URL of the downloaded attachment.
+    public let localFileURL: URL
+    
+    /// The local state of the attachment.
+    public let state: LocalAttachmentState
+    
+    /// The information about file size/mimeType.
+    public let file: AttachmentFile
+}
+
```

Comment:
Similar to the `AttachmentUploadingState` with one downside, localFileURL is only valid when state == .downloaded. Should it be optional?

Reply:
Yes, that would be better